presto-main/src/test/java/io/prestosql/sql/planner/TestDomainCoercer.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.sql.planner;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.metadata.Metadata;
+import io.prestosql.spi.predicate.Domain;
+import io.prestosql.spi.predicate.ValueSet;
+import io.prestosql.spi.type.Type;
+import io.prestosql.spi.type.TypeOperators;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static io.prestosql.SessionTestUtils.TEST_SESSION;
+import static io.prestosql.metadata.MetadataManager.createTestMetadataManager;
+import static io.prestosql.spi.predicate.Domain.multipleValues;
+import static io.prestosql.spi.predicate.Range.greaterThan;
+import static io.prestosql.spi.predicate.Range.greaterThanOrEqual;
+import static io.prestosql.spi.predicate.Range.lessThan;
+import static io.prestosql.spi.predicate.Range.range;
+import static io.prestosql.spi.type.BigintType.BIGINT;
+import static io.prestosql.spi.type.DecimalType.createDecimalType;
+import static io.prestosql.spi.type.DoubleType.DOUBLE;
+import static io.prestosql.spi.type.IntegerType.INTEGER;
+import static io.prestosql.spi.type.RealType.REAL;
+import static io.prestosql.spi.type.SmallintType.SMALLINT;
+import static org.testng.Assert.assertEquals;
+
+public class TestDomainCoercer
+{
+    private Metadata metadata;
+    private TypeOperators typeOperators;
+
+    @BeforeClass
+    public void setup()
+    {
+        metadata = createTestMetadataManager();
+        typeOperators = new TypeOperators();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void tearDown()
+    {
+        metadata = null;
+    }
+
+    private Domain applySaturatedCasts(Domain domain, Type coercedValueType)
+    {
+        return DomainCoercer.applySaturatedCasts(metadata, typeOperators, TEST_SESSION, domain, coercedValueType);
+    }
+
+    @Test
+    public void testNone()
+    {
+        assertEquals(applySaturatedCasts(Domain.none(BIGINT), INTEGER), Domain.none(INTEGER));
+    }
+
+    @Test
+    public void testAll()
+    {
+        assertEquals(applySaturatedCasts(Domain.all(BIGINT), INTEGER), Domain.all(INTEGER));
+    }
+
+    @Test
+    public void testOnlyNull()
+    {
+        assertEquals(applySaturatedCasts(Domain.onlyNull(BIGINT), INTEGER), Domain.onlyNull(INTEGER));
+    }
+
+    @Test
+    public void testCoercedValueSameAsOriginal()
```

Comment:
Is not-same as original tested anywhere? Preferably we should have a test case where there is a range that can be coerced and an exact value that cannot be coerced.

Reply:
testTruncatedCoercedValue has a case where we coerce `[123.456, 234.567]` to `(123.4, max)`